draco/core/misc.py

```diff
@@ -214,6 +217,36 @@ def finish(self):
         return items
 
 
+class CheckMPIEnvironment(task.MPILoggedTask):
+    """Check that the current MPI environment can communicate
+    across all nodes.
+    """
+
+    timeout = config.Property(proptype=int, default=60)
```

Comment:
This might not be long enough. The job start up time can often be a few minutes, and you might imagine that could vary a lot between ranks. If one starts up a minute before the others then this will fail regardless of whether they could all communicate.